staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -104,6 +104,19 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	return ev, nil
 }
 
+// UpdateEventAttribs updates authorizer attributes according to attribs
+func UpdateEventAttribs(ev *auditinternal.Event, attribs authorizer.Attributes) {
+	if user := attribs.GetUser(); user != nil {
+		ev.User.Username = user.GetName()
```

Comment:
Hi, @tallclair  
I think we can extend the event type, add a new ev.EffectiveUser to events.

The ev.User in the log is the "real" user, but it is not the "effective" user. Because impersonation will change the user info after the audit filter gathers event info.

Can we add this to the [TODO list](https://github.com/kubernetes/kubernetes/issues/48561) @tallclair ?

@ericchiang What's your opinion?